### PR TITLE
Feature: add easter egg system

### PR DESCRIPTION
**Pull request by Ratismal**
Adds a system that allows easter eggs to be defined within the app. Easter eggs (referred to as "eggs"), can be defined in an `eggs.json` file with the following schema:
```jsonc
[
  // Each object in the array is an egg
  {
    // The 'match' property contains a list of matchers. If any of them resolve as true, the egg is considered to be matched.
    // All terms are compared to a lowercased seed, and thus should also be provided in lowercase.
    // See the 'matchers' object in eggs.js for a full list
    "match": {
      "all": ["terms"], // the seed must contain all the listed terms
      "any": ["terms"], // the seed must contain any of the listed terms
      "exact": ["terms"] // the seed must exactly be any of the listed terms
    },
    // The 'mutate' property contains a list of mutations. All of them get executed in order. 
    // See the 'mutations' object in eggs.js for a full list
    "mutate": {
      "background": "red", // Sets the background property of the body element
      "css": "body { background: 'red'; }" // Adds a new style tag with the given content
    }
  }
]
```

Eggs are matched against the current page's seed, forced to lowercased. Multiple eggs can match for a given seed, which could lead to interesting (or deadly) combinations. The list of matchers and mutations can be expanded at the contributor's discretion.

### CODE REVIEW

**Review by queer — CHANGES_REQUESTED** [review]
Literally unreadable

**Comment by Ratismal** [comment]
im going to cry

**Review by queer — APPROVED** [review]

File: eggs.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+			}
+		}
+
+		return true;
+	},
+
+	// Returns true if the seed includes all terms
+	any(terms, seed) {
+		for (const term of terms) {
+			if (!seed.includes(term)) {
```
Comment (remram44) [inline eggs.js]: Are you sure about the `!`?

Comment (Ratismal) [reply eggs.js]: yup you're right, good catch. copy paste error from matchers#all
```suggestion
			if (seed.includes(term)) {
```

File: eggs.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+				return false;
+			}
+		}
+
+		return true;
+	},
+
+	// Returns true if the seed includes all terms
```
Comment (Ratismal) [inline eggs.js]:
```suggestion
	// Returns true if the seed includes any term
```

**Pull request merged**